### PR TITLE
Jupiter 10.12.0 e cdn

### PR DESCRIPTION
**Pull request by yossipapi**

### CODE REVIEW

File: alpha/lib/enums/PermissionName.php [hunk 1]
```diff
@@ -241,4 +241,10 @@
 	const FEATURE_SHOW_FLASH_STUDIO = 'FEATURE_SHOW_FLASH_STUDIO';
 	const FEATURE_SHOW_HTML_STUDIO = 'FEATURE_SHOW_HTML_STUDIO';
 	const FEATURE_HIDE_TEMPLATE_PARTNER_UICONFS = 'FEATURE_HIDE_TEMPLATE_PARTNER_UICONFS';
+
```
Comment (tan-tan-kanarek) [inline alpha/lib/enums/PermissionName.php]: Why did you add these constants, they're never used.

File: alpha/config/schema.xml [hunk 2]
```diff
@@ -1720,6 +1720,26 @@
 		<column primaryKey="false" required="false" autoIncrement="false" lazyLoad="false" name="delivery_priority" type="integer" />
 		<column primaryKey="false" required="false" autoIncrement="false" lazyLoad="false" name="delivery_status" type="integer" />
 	</table>
+	<table skipSql="false" abstract="false" name="edge_server">
```
Comment (tan-tan-kanarek) [inline alpha/config/schema.xml]: I think you better add type column, we will use it one day.

File: alpha/lib/model/EdgeServer.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * long as it does not already exist in the output directory.
+ *
+ * @package Core
+ * @subpackage model
+ */
+class EdgeServer extends BaseEdgeServer {
+	
+	const CUSTOM_DATA_DELIVERY_IDS = 'delivery_profile_ids';
+
+	/**
```
Comment (tan-tan-kanarek) [inline alpha/lib/model/EdgeServer.php]: The constructor is redundant.

File: api_v3/lib/types/edgeServer/KalturaEdgeServer.php [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+				throw new KalturaAPIException(KalturaErrors::SYSTEM_NAME_ALREADY_EXISTS, $this->systemName);
+		}
+		
+		return parent::validateForUpdate($sourceObject, $propertiesToSkip);
+	}
+	
+	public function validateMandatoryAttributes()
+	{
+		$this->validatePropertyMinLength("name", 1, true);
+		$this->validatePropertyNotNull("hostName");
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/edgeServer/KalturaEdgeServer.php]: It's called also from validateForUpdate, you should allow null in such case.

File: api_v3/lib/types/edgeServer/KalturaEdgeServer.php [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+		if($deliveryProfileIds)
+			foreach($deliveryProfileIds->toArray() as $keyValue) 
+				$this->insertObject($deliveryProfiles, $keyValue->key, $keyValue->value);
+			
+		$object_to_fill->setDeliveryProfileIds($deliveryProfiles);
+		
+		return $object_to_fill;
+	}
+	
+	protected function insertObject(&$res, $key, $value) {
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/edgeServer/KalturaEdgeServer.php]: You define an internal language here, I think you better use well structured API object instead.

File: api_v3/lib/types/edgeServer/KalturaEdgeServerStatus.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?php
+/**
+ * @package api
+ * @subpackage enum
+ */
+class KalturaEdgeServerStatus extends KalturaEnum
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/edgeServer/KalturaEdgeServerStatus.php]: Please implement EdgeServerStatus instead of redefining that constants.

File: api_v3/lib/types/edgeServer/KalturaEdgeServer.php (hunk 5 above)
Comment (yossipapi) [inline api_v3/lib/types/edgeServer/KalturaEdgeServer.php]: changed to comma separated ids so skipping this comment

File: api_v3/lib/types/edgeServer/KalturaEdgeServerStatus.php (hunk 6 above)
Comment (yossipapi) [inline api_v3/lib/types/edgeServer/KalturaEdgeServerStatus.php]: Done

File: alpha/lib/model/EdgeServer.php (hunk 3 above)
Comment (yossipapi) [inline alpha/lib/model/EdgeServer.php]: The Ctor was auto generated so I did not want to remove it to not create future diffs.

File: alpha/config/schema.xml (hunk 2 above)
Comment (yossipapi) [inline alpha/config/schema.xml]: Done + added new type edgeServerType::NODE
